Assets/Scripts/FloatingOrigin/Origin.cs

```diff
@@ -53,25 +53,25 @@ private IEnumerator AttemptShift()
             {
                 yield return new WaitForEndOfFrame();
 
-                var mainShifterPosition = mainShifter.position;
-                var originPosition = transform.position;
-                var newPosition = new Vector3(mainShifterPosition.x, originPosition.y, mainShifterPosition.z);
-                
-                // We use newPosition instead of mainShifterPosition so that the Y component is the same and we 
-                // calculate the distance on a flat plane. This prevents origin shifts when the only change of the
-                // shifter is the height.
-                var distanceBetweenMainShifterAndOrigin = newPosition - originPosition;
-
-                // sqrMagnitude is used because it outperforms magnitude quite a bit: https://docs.unity3d.com/ScriptReference/Vector3-sqrMagnitude.html
-                if (distanceBetweenMainShifterAndOrigin.sqrMagnitude < sqrDistanceBeforeShifting)
+                if (((mainShifter.position.x-transform.position.x)* (mainShifter.position.x - transform.position.x))+ ((mainShifter.position.z - transform.position.z) * (mainShifter.position.z - transform.position.z))< sqrDistanceBeforeShifting)
                 {
                     continue;
                 }
 
+
+                var mainShifterPosition = mainShifter.position;
+                Coordinate mainShifterCoordinate = new Coordinate(mainShifterPosition);
+                Coordinate mainShifterWGS = mainShifterCoordinate.Convert(CoordinateSystem.WGS84_LatLonHeight);
```

Comment:
Is deze conversie nodig? Of moet eigenlijk de applicatie CRS op CoordinateSystem.WGS84_LatLonHeight staan?

Reply:
Dat zou mogelijk performance schelen namelijk omdat er minder berekeningen gedaan hoeven te worden in een shift

Reply:
de applicatie-CRS kan niet ingesteld worden worden op een crs met coordinaten in graden, omdat lengte en breedtegraden niet direct mappen naar meters.

de conversie naar wgs84 wordt alleen uitgevoerd wanneer de origin daadwerkelijk moet verschuiven, dus ik denk niet dat dat veel invloed heeft op de performance.

het idee achter het converteren naar wgs84 is, dat de origin altijd op dezelfde hoogte boven de wgs84-geoide blijft. wanneer je de hoogte in het gekoppelde crs gebruikt (vooral als dit een ecef-CRS is) komt je origin al snel heel ver boven of onder je terreinoppervlakte. (bij ecef is de hoogte-as loodrecht op het vlak van de evenaar richting de noordpool, 1000 m naar het noorden is dan (in nederland) ook ca 1000 meter omhoog.

Reply:
heb wat comments toegevoegd om het achterliggende idee toe te lichten